### PR TITLE
Flywoo H743 Pro

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ - IMU - Dual ICM42688
+ - Barometer - SPL06
+ - OSD - AT7456E
+ - Onboard Flash: 500MByte
+ - 7x UARTs
+ - 13x PWM Outputs (12 Motor Output, 1 LED)
+ - Battery input voltage: 2S-6S
+ - BEC 3.3V 0.5A
+ - BEC 5V 3A
+ - BEC 10V 3A
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:17]:
```suggestion
 - BEC 10V 3A for video, gpio controlled
 - Dual switchable camera inputs
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (RX/SBUS, DMA-enabled)
+ - SERIAL2 -> UART2 (GPS, DMA-enabled)
+ - SERIAL3 -> UART3
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:32]:
```suggestion
 - SERIAL3 -> UART3 (User)
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ - SERIAL4 -> UART4 (GPS, DMA-enabled)
+ - SERIAL6 -> UART6 (ESC Telemetry)
+ - SERIAL7 -> UART7
+ - SERIAL8 -> UART8 (DJI/MSP, DMA-enabled)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:36]:
```suggestion
 - SERIAL8 -> UART8 (DisplayPort, DMA-enabled)
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (RX/SBUS, DMA-enabled)
+ - SERIAL2 -> UART2 (GPS, DMA-enabled)
+ - SERIAL3 -> UART3
+ - SERIAL4 -> UART4 (GPS, DMA-enabled)
+ - SERIAL6 -> UART6 (ESC Telemetry)
+ - SERIAL7 -> UART7
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:35]:
```suggestion
 - SERIAL7 -> UART7 (User)
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ - SERIAL8 -> UART8 (DJI/MSP, DMA-enabled)
+
+## RC Input
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:38]: replace RC section with this:
RC input is configured by default via the USART1 RX input. It supports all unidirectional RC protocols except PPM. 

Note: If the receiver is FPort the receiver must be tied to the USART1 TX pin ,  RSSI_TYPE set to 3, and SERIAL2_OPTIONS must be set to 7 (invert TX/RX, half duplex). For full duplex like CRSF/ELRS use both RX1 and TX1 and set RSSI_TYPE also to 3.

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+## FrSky Telemetry
+ 
+FrSky Telemetry is supported using the T5 pin (UART5 transmit). You need to set the following parameters to enable support for FrSky S.PORT
+ 
+  - SERIAL5_PROTOCOL 10
+  - SERIAL5_OPTIONS 7
+  
+## OSD Support
+
+The Flywoo H743 Pro supports OSD using OSD_TYPE 1 (MAX7456 driver).
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:54]:
```suggestion
The Flywoo H743 Pro supports OSD using OSD_TYPE 1 (MAX7456 driver) and simultaneously DisplayPort using TX8/RX8 on the HD VTX connector.
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+ - PWM 11-12 in group4
+ - PWM 13    in group5
+
+Channels within the same group need to use the same output rate. If
+any channel in a group uses DShot then all channels in the group need
+to use DShot. Channels 1-10 support bi-directional dshot.
+
+## Battery Monitoring
+
+The board has a builting voltage and current sensor. The current
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:76]:
```suggestion
The board has a built-in voltage sensor and external current sensor input. The current
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+## Loading Firmware
+
+Initial firmware load can be done with DFU by plugging in USB with the
+bootloader button pressed. Then you should load the "with_bl.hex"
+firmware, using your favourite DFU loading tool.
+
+Once the initial firmware is loaded you can update the firmware using
+any ArduPilot ground station software. Updates should be done with the
+*.apj firmware files.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:100]:
```suggestion
\*.apj firmware files.
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+## RC Input
+
+RC input is configured on the R1 (UART1_RX) pin. It supports all serial RC
+protocols. For protocols requiring half-duplex serial to transmit
+telemetry (such as FPort) you should setup SERIAL6 as an RC input serial port,
+with half-duplex, pin-swap and inversion enabled.
+ 
+## FrSky Telemetry
+ 
+FrSky Telemetry is supported using the T5 pin (UART5 transmit). You need to set the following parameters to enable support for FrSky S.PORT
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:47]:
```suggestion
FrSky Telemetry is supported using  and unused UART,such as the T3 pin (UART3 transmit). You need to set the following parameters to enable support for FrSky S.PORT
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+ 
+  - SERIAL5_PROTOCOL 10
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:49]:
```suggestion
  - SERIA3_PROTOCOL 10
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  - SERIAL5_OPTIONS 7
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:50]:
```suggestion
  - SERIAL3_OPTIONS 7
```

File: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+  
+## OSD Support
+
+The Flywoo H743 Pro supports OSD using OSD_TYPE 1 (MAX7456 driver).
+
+## PWM Output
+
+The Flywoo H743 Pro supports up to 13 PWM outputs. The pads for motor output
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md:58]:
```suggestion
The Flywoo H743 Pro supports up to 13 PWM or DShot outputs. The pads for motor output
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
need  to add readme sections on VTX power control and camera switch also

**Comment by andyp1per** [comment]
All fixed, @Hwurzburg

**Comment by Hwurzburg** [comment]
@andyp1per  missed a few

**Comment by andyp1per** [comment]
Arg, github and its hidden conversations! Fixed now.

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**